bedrock/contentful/api.py

```diff
@@ -455,92 +456,114 @@ def render_rich_text(self, node):
-        blurb = fields.get("preview_blurb", "")
+        title = entry_fields.get("preview_title", title)
+        blurb = entry_fields.get("preview_blurb", "")
 
         if seo_fields:
             # Defer to SEO fields for blurb if appropriate.
             blurb = seo_fields.get("description", "")
 
-        campaign = f"{in_firefox}{slug}"
-        page_type = entry_obj.content_type.id
+        return slug, title, blurb
+
+    def _get_info_data__category_tags_classification(self, entry_fields, page_type):
 
+        data = {}
+
+        # TODO: Check with plans for Contentful use - we may
+        # be able to relax this check and use it for page types
+        # once we're in all-Compose mode
+        if page_type == CONTENT_TYPE_PAGE_RESOURCE_CENTER:
+            if "category" in entry_fields:
+                data["category"] = entry_fields["category"]
+            if "tags" in entry_fields:
+                data["tags"] = entry_fields["tags"]
+            if "product" in entry_fields:
+                # NB: this is a re-mapping with an eye on flexibility - pages may not always have
+                # a 'product' key, but they might have something regarding overall classification
+                data["classification"] = entry_fields["product"]
+        return data
+
+    def _get_info_data__theme_campaign(self, entry_fields, slug):
+        _folder = entry_fields.get("folder", "")
+        _in_firefox = "firefox-" if "firefox" in _folder else ""
+        campaign = f"{_in_firefox}{slug}"
+        theme = "firefox" if "firefox" in _folder else "mozilla"
+        return theme, campaign
+
+    def _get_info_data__locale(self, page_type, entry_fields, entry_obj):
         # TODO: update this once we have a robust locale field available (ideally
         # via Compose's parent `page`), because double-purposing the "name" field
         # is a bit too brittle.
         if page_type == "pageHome":
-            locale = fields["name"]
+            locale = entry_fields["name"]
         else:
             locale = entry_obj.sys["locale"]
+        return locale
 
-        data.update(
-            {
-                "title": title,
-                "blurb": blurb,
-                "slug": slug,
-                "locale": locale,
-                "theme": "firefox" if "firefox" in folder else "mozilla",
-                # eg www.mozilla.org-firefox-accounts or www.mozilla.org-firefox-sync
-                "utm_source": f"www.mozilla.org-{campaign}",
-                "utm_campaign": campaign,  # eg firefox-sync
-            }
-        )
+    def get_info_data(self, entry_obj, seo_obj=None):
+        # TODO, need to enable connectors
+        entry_fields = entry_obj.fields()
+        if seo_obj:
+            seo_fields = seo_obj.fields()
+        else:
+            seo_fields = None
+
+        page_type = entry_obj.content_type.id
+
+        slug, title, blurb = self._get_info_data__slug_title_blurb(entry_fields, seo_fields)
```

Comment:
It looks like you could simplify things just a bit by having all of these new methods return dicts of data instead of tuples. The first one becomes `data`, then the following ones update `data` just like the last one does. I don't see any further processing of these data, so the new varaibles don't seem valuable. Just a style nit as this works fine. Feel free to take or leave the suggestion.

Reply:
Yeah, fair call - was thinking something similar earlier but got to fixed on getting this to PR. On it now

Reply:
`campaign` does get double-used, but that's an easy juggle

Reply:
yeah, good point. the method could return those two dict keys as well, or we can use the `data["campaign"]` there. Your call.